src/ip_grid_descriptor_mod.f90

```diff
@@ -1,3 +1,13 @@
+!> @file
+!! @brief Uses derived type grid descriptor objects to abstract away the raw Grib-1 and Grib-2 grid definitions.
+!! @author Kyle Gerheiser
+
+!> @brief Uses derived type grid descriptor objects to abstract away the raw Grib-1 and Grib-2 grid definitions.
+!! @detail
+!! An abstract ip_grid_descriptor class is subclassed by the Grib-1 and Grib-2 descriptor types which contain
+!! the raw integer grib arrays. A comparison operator to test whether two grid descriptors are the same.
```

Comment:
Can you properly line-break in your comments to keep them under 80 chars long?

Emacs has a command fill-paragraph which does this. Your editor no doubt also has this capability.